serial/tools/list_ports_linux.py

```diff
@@ -90,11 +90,20 @@ def read_line(self, *args):
 
 def comports(include_links=False, hide_subsystems=["platform"]):
     devices = set()
-    drivers = open('/proc/tty/drivers').readlines()
-    for driver in drivers:
-        items = driver.strip().split()
-        if items[4] == 'serial':
-            devices.update(glob.glob(items[1]+'*'))
+    for driver in open('/proc/tty/drivers'):
+        items = driver.readlines().strip().split()
+        try:
+            if items[4] == 'serial':
```

Comment:
We can add a length test before accessing column 4.
`if len(items) > 4 and items[4] == 'serial':`

Reply:
I could, but assuming the file was messed up (or in some uncommon distribution or something), we would end up adding no ports, and code that previously worked would fail. I want to avoid introducing any regressions, hence falling back to the old functionality in unforeseen circumstances.